doc/contribute/external.rst

```diff
@@ -103,6 +103,44 @@ With this approach the code is considered as being developed externally, and
 thus it is not automatically subject to the requirements of the previous
 section.
 
+Integration in main manifest file (west.yaml)
++++++++++++++++++++++++++++++++++++++++++++++
+
+Integrating external code into the main :file:`west.yml` manifest file is
+limited to code that is used by a Zephyr subsystem (libraries), by a platform,
+drivers (HAL) or tooling needed to test or build Zephyr components.
+
+The integration of modules in this group is validated by the Zephyr project CI,
+and verified to be working with each Zephyr release.
```

Comment:
Can you add language similar to what was discussed in the Process WG. https://github.com/zephyrproject-rtos/zephyr/issues/46135#issuecomment-1717990887

Specifically some language that indicates "integrated modules will not be removed from the tree without a detailed migration plan".